src/debugging/coreclr/CommandLineDotNetClient.ts

```diff
@@ -48,6 +60,75 @@ export class CommandLineDotNetClient implements DotNetClient {
             return undefined;
         }
     }
+
+    public async trustAndExportSslCertificate(projectFile: string, hostExportPath: string, containerExportPath: string): Promise<void> {
+        if (CommandLineDotNetClient.KnownConfiguredProjects.has(projectFile)) {
+            return;
+        }
+
+        await this.addUserSecretsIfNecessary(projectFile);
+        await this.promptAndTrustCertificateIfNecessary();
+
+        const password = uuidv4();
+
+        // Export the certificate
+        const exportCommand = `dotnet dev-certs https -ep "${hostExportPath}" -p "${password}"`;
+        await this.processProvider.exec(exportCommand, {});
+
+        // Set the password to dotnet user-secrets
+        const userSecretsPasswordCommand = `dotnet user-secrets --project "${projectFile}" set Kestrel:Certificates:Development:Password "${password}"`;
+        await this.processProvider.exec(userSecretsPasswordCommand, {});
+
+        // This is not honored due to https://github.com/aspnet/AspNetCore.Docs/issues/6199#issuecomment-418194220
+        // Consequently, the certificate name must be equal to <binaryName>.pfx, i.e. MyWebApp.dll => MyWebApp.pfx
+        //const userSecretsPathCommand = `dotnet user-secrets --project "${projectFile}" set Kestrel:Certificates:Development:Path "${containerExportPath}"`;
```

Comment:
I wrestled with that. We've had endless problems with paths when they contain spaces, unicode characters, etc. in VS. I would have strongly preferred explicitly setting the path to use so it doesn't depend on the shaky logic of "MyWebApp.dll" => "MyWebApp.pfx", especially considering the logic that we use to determine the DLL name is itself rather shaky.

I'll remove it, but if they ever fix that bug this would be a point of improvement for us.